rust/alloc/raw_vec.rs

```diff
@@ -218,6 +235,29 @@ impl<T, A: Allocator> RawVec<T, A> {
         }
     }
 
+    fn try_allocate_in(capacity: usize, init: AllocInit, alloc: A) -> Result<Self, TryReserveError> {
+        if mem::size_of::<T>() == 0 {
+            return Ok(Self::new_in(alloc));
+        }
+
+        let layout = Layout::array::<T>(capacity)?;
+        alloc_guard(layout.size())?;
+        let result = match init {
+            AllocInit::Uninitialized => alloc.allocate(layout),
+            AllocInit::Zeroed => alloc.allocate_zeroed(layout),
+        };
+        let ptr = match result {
+            Ok(ptr) => ptr,
+            Err(_) => return Err(TryReserveError::AllocError{ layout, non_exhaustive: () }),
```

Comment:
It looks like there is a spacing issue here: should be `AllocErr {`. I think probably it's easier to make a change in a Rust clone first, and then use `./x.py fmt`, and then transfer the file to kernel clone.

Reply:
Good catch, thanks! Formatting is not too critical since these should eventually go upstream, but yeah, we should do a pass from time to time to see if we are missing something. Same applies to Clippy, Miri, checking the doctests, etc.